### PR TITLE
fix: Add Cache-Control: no-store to render_template function

### PR DESCRIPTION
**Pull request by frankh**
## Problem
Our web app responses should not be cached as they can contain private information that can be retrieved by e.g. hitting the back button after logging out

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/utils.py [hunk 1]
```diff
@@ -418,7 +419,7 @@ def render_template(
     context["posthog_js_uuid_version"] = settings.POSTHOG_JS_UUID_VERSION
 
     html = template.render(context, request=request)
-    return HttpResponse(html)
+    return patch_cache_control(HttpResponse(html), no_store=True)
```
Comment (webjunkie) [inline posthog/utils.py:422]: If we never want to cache, there's `@never_cache`

> from django.views.decorators.cache import never_cache
>
>
> @never_cache
> def myview(request): ...

**Review by webjunkie — COMMENTED** [review]
Flyby

File: posthog/utils.py (hunk 1 above)
Comment (Twixes) [reply posthog/utils.py:422]: Code quality checks are right here.

One more thing: I believe we should only prevent caching for logged-in requests, i.e. `not request.user.is_anonymous`. So something like this:
```suggestion
    response = HttpResponse(html)
    if not user.request.is_anonymous:
        patch_cache_control(response, no_store=True)
    return response
```

**Review by Twixes — APPROVED** [review]
👌

**Review by webjunkie — APPROVED** [review]

**Comment by raquelmsmith** [comment]
Originally reported by smit@esecurify.com - he requested recognition and we're happy to give it!

**Pull request merged**